### ISSUE TITLE
Hemes + Instabug NetworkLogger = 💥 

### ISSUE THREAD

**Issue by retyui** [Instabug/Instabug-React-Native#878]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `instabug-reactnative@10.11.0` for the project I'm working on.

Hermes is default engine and `eval` didn't work as expected , can we make a hot fix for 10.x.x?

```tsx
NetworkLogger.setRequestFilterExpression('network.url.includes("generate_204")')

// ReferenceError: Property 'network' doesn't exist
```

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/instabug-reactnative/modules/NetworkLogger.js b/node_modules/instabug-reactnative/modules/NetworkLogger.js
index 88f6019..fb47125 100644
--- a/node_modules/instabug-reactnative/modules/NetworkLogger.js
+++ b/node_modules/instabug-reactnative/modules/NetworkLogger.js
@@ -5,7 +5,7 @@ import InstabugConstants from '../utils/InstabugConstants';
 let { Instabug, IBGAPM } = NativeModules;
 
 var _networkDataObfuscationHandlerSet = false;
-var _requestFilterExpression = false;
+var _requestFilterExpression = ()=>true;
 
 /**
  * NetworkLogger
@@ -21,7 +21,7 @@ export default {
     if (isEnabled) {
       xhr.enableInterception();
       xhr.setOnDoneCallback(network => {
-        if (!eval(_requestFilterExpression)) {
+        if (!_requestFilterExpression(network)) {
           if (_networkDataObfuscationHandlerSet) {
             IBGEventEmitter.emit(
               InstabugConstants.NETWORK_DATA_OBFUSCATION_HANDLER_EVENT,
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by TheBuggedYRN** [comment]
Thanks for reporting this, @retyui!
This issues was fixed in #705 which had been released since v11.0.0. You need to update to the latest version by running:

```bash
# npm
npm install instabug-reactnative@latest

# yarn
yarn upgrade instabug-reactnative --latest
```

Since you current version is `10.x.x`, updating to the latest version will have some breaking changes. Here are some links that will make the migration smooth:
1. [SDK v11.0.0 Migration Guide](https://docs.instabug.com/docs/react-native-migration-guide)
1. [Changelog](https://github.com/Instabug/Instabug-React-Native/blob/master/CHANGELOG.md)

**Comment by retyui** [comment]
<img width="776" alt="Screenshot 2023-01-06 at 11 25 57" src="https://user-images.githubusercontent.com/4661784/210986636-80a51569-9b74-4e16-a169-82e42af381f5.png">

migration take time to cover all breaking changes, 10.x still the most popular version 

can you make a hotfix for `10.x.x`?

**Comment by TheBuggedYRN** [comment]
Unfortunately, we no longer support `v10.x.x`.

Migrating to `v11.x.x` might seem like a long process, but as long as you do not use any deprecated API, things will go smooth after [moving the initialization code from native-side to JS-side](https://docs.instabug.com/docs/react-native-migration-guide#migrate-the-initialization-code-to-js). There has been a lot of bug fixes and improvements since `v11.0.0` so it's always a good practice to keep the SDK up to date.

Don't hesitate to reach out if you need any help with the migration process.
We are always happy to help!
